src/main/scala/cognite/spark/v1/RelationshipsRelation.scala

```diff
@@ -0,0 +1,156 @@
+package cognite.spark.v1
+
+import java.time.Instant
+
+import cats.effect.IO
+import cats.implicits._
+import cognite.spark.v1.PushdownUtilities.{
+  confidenceRangeFromMinAndMax,
+  getLabelsFilter,
+  idsFromWrappedArray,
+  pushdownToParameters,
+  shouldGetAll,
+  stringSeqFromWrappedArray,
+  timeRangeFromMinAndMax,
+  toPushdownFilterExpression
+}
+import cognite.spark.v1.SparkSchemaHelper.{asRow, fromRow, structType}
+import com.cognite.sdk.scala.v1._
+import fs2.Stream
+import org.apache.spark.sql.sources.{Filter, InsertableRelation}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Row, SQLContext}
+
+class RelationshipsRelation(config: RelationConfig)(val sqlContext: SQLContext)
+    extends SdkV1Relation[Relationship, String](config, "relationships")
+    with InsertableRelation
+    with WritableRelation {
+  import CdpConnector._
+
+  override def schema: StructType = structType[Relationship]
+
+  override def toRow(a: Relationship): Row = asRow(a)
+
+  override def uniqueId(a: Relationship): String = a.externalId
+
+  override def getStreams(filters: Array[Filter])(
+      client: GenericClient[IO],
+      limit: Option[Int],
+      numPartitions: Int): Seq[Stream[IO, Relationship]] = {
+    val fieldNames =
+      Array(
+        "sourceExternalId",
+        "sourceType",
+        "subtype",
+        "targetExternalId",
+        "targetType",
+        "minStartTime",
+        "maxStartTime",
+        "minEndTime",
+        "maxEndTime",
+        "minActiveAtTime",
+        "maxActiveAtTime",
+        "minConfidence",
+        "maxConfidence",
+        "dataSetId",
+        "labels",
+        "minCreatedTime",
+        "maxCreatedTime",
+        "minLastUpdatedTime",
+        "maxLastUpdatedTime"
+      )
+    val pushdownFilterExpression = toPushdownFilterExpression(filters)
+    val shouldGetAllRows = shouldGetAll(pushdownFilterExpression, fieldNames)
+    val filtersAsMaps = pushdownToParameters(pushdownFilterExpression)
+
+    val relationshipsFilterSeq = if (filtersAsMaps.isEmpty || shouldGetAllRows) {
+      Seq(RelationshipsFilter())
+    } else {
+      filtersAsMaps.distinct.map(relationshipsFilterFromMap)
+    }
+
+    val streamsPerFilter = relationshipsFilterSeq
+      .map { f =>
+        client.relationships.filterPartitions(f, numPartitions, limit)
+      }
+
+    // Merge streams related to each partition to make sure duplicate values are read into
+    // the same RDD partition
+    streamsPerFilter.transpose
+      .map(s => s.reduce(_.merge(_)))
+  }
+
+  def relationshipsFilterFromMap(m: Map[String, String]): RelationshipsFilter =
+    RelationshipsFilter(
+      sourceExternalIds = m.get("sourceExternalIds").map(stringSeqFromWrappedArray),
+      sourceTypes = m.get("sourceTypes").map(stringSeqFromWrappedArray),
+      targetExternalIds = m.get("targetExternalIds").map(stringSeqFromWrappedArray),
```

Comment:
As far as I understand, we need a function to evaluate strings here meaning "array('ext1', 'ext2')" or "array(\"ext1\", \"ext2\")" into Seq("ext1", "ext2") - just thinking about array function I used in transformations. 
Is there a builtin function for that?

Reply:
Also not sure about this question, but that's what the `stringSeqFromWrappedArray` is doing already so we're good(?)

Reply:
No, I created that function (as a placeholder) but it is not implemented properly :D

Reply:
I think it should be good actually, given that `idsFromWrappedArray` works :)

Reply:
Spent some time but could not do it properly. (very dumb when it comes to regex maybe :D )

Reply:
ah, I checked that one. It works for lists of integers but not for lists of strings. In relationships, we receive a list of external ids instead of ids. So we need a different function to parse it.